Library doesn't compile with latest Mbed board software version 1.3.0

When I try and compile the shining keyboard example on an MBED board running version 1.3.0 I get the following errors:

```
In file included from /Users/bsperryn/Documents/Arduino/shining_keyboard/shining_keyboard.ino:12:0:
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/signal_utils.h:55:7: warning: 'float {anonymous}::randf(float, float)' defined but not used [-Wunused-function]
 float randf(float edge0=0.0f, float edge1=1.0f) {
       ^~~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/signal_utils.h:39:7: warning: 'float {anonymous}::mmap(float, float, float, float, float)' defined but not used [-Wunused-function]
 float mmap(float x, float a, float b, float c, float d) {
       ^~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/signal_utils.h:17:7: warning: 'float {anonymous}::step(float, float)' defined but not used [-Wunused-function]
 float step(float a, float x) {
       ^~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp: In static member function 'static void MbedBleHID::RunEventThread(void (*)())':
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:46:36: warning: 'int events::EventQueue::call_every(int, Mbed_F) [with Mbed_F = void (*)()]' is deprecated: Pass a chrono duration, not an integer millisecond count. For example use `5s` rather than `5000`. [since mbed-os-6.0.0] [-Wdeprecated-declarations]
   eventQueue.call_every(10, task_fn);
                                    ^
In file included from /Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed/events/mbed_events.h:21:0,
                 from /Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed/mbed.h:32,
                 from /Users/briansperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed.h:14,
                 from /Users/briansperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.h:7,
                 from /Users/briansperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:2:
/Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed/events/EventQueue.h:1063:9: note: declared here
     int call_every(int ms, F f)
         ^~~~~~~~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:53:44: warning: 'void rtos::ThisThread::sleep_for(uint32_t)' is deprecated: Pass a chrono duration, not an integer millisecond count. For example use `5s` rather than `5000`. [since mbed-os-6.0.0] [-Wdeprecated-declarations]
   rtos::ThisThread::sleep_for(osWaitForever);
                                            ^
In file included from /Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed/rtos/rtos.h:30:0,
                 from /Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/USB/PluggableUSBSerial.h:24,
                 from /Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/Arduino.h:131,
                 from /Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed.h:4,
                 from /Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.h:7,
                 from /Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:2:
/Users/bsperryn/Library/Arduino15/packages/arduino/hardware/mbed/1.3.0/cores/arduino/mbed/rtos/ThisThread.h:216:6: note: declared here
 void sleep_for(uint32_t millisec);
      ^~~~~~~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp: In member function 'void MbedBleHID::postInitialization(BLE&)':
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:105:14: error: 'ConnectionParams_t' in 'Gap {aka class ble::interface::Gap<ble::generic::GenericGap<ble::pal::vendor::cordio::Gap, ble::pal::SecurityManager<ble::pal::vendor::cordio::CordioSecurityManager<ble::generic::GenericSecurityManager<ble::pal::vendor::cordio::CordioSecurityManager, ble::vendor::cordio::SigningEventMonitor> >, ble::generic::GenericSecurityManager<ble::pal::vendor::cordio::CordioSecurityManager, ble::vendor::cordio::SigningEventMonitor> >, ble::generic::GenericSecurityManager<ble::pal::vendor::cordio::CordioSecurityManager, ble::vendor::cordio::SigningEventMonitor> > >}' does not name a type
   const Gap::ConnectionParams_t connectionParams = {
              ^~~~~~~~~~~~~~~~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:111:7: error: 'Gap {aka class ble::interface::Gap<ble::generic::GenericGap<ble::pal::vendor::cordio::Gap, ble::pal::SecurityManager<ble::pal::vendor::cordio::CordioSecurityManager<ble::generic::GenericSecurityManager<ble::pal::vendor::cordio::CordioSecurityManager, ble::vendor::cordio::SigningEventMonitor> >, ble::generic::GenericSecurityManager<ble::pal::vendor::cordio::CordioSecurityManager, ble::vendor::cordio::SigningEventMonitor> >, ble::generic::GenericSecurityManager<ble::pal::vendor::cordio::CordioSecurityManager, ble::vendor::cordio::SigningEventMonitor> > >}' has no member named 'setPreferredConnectionParams'; did you mean 'PreferredConnectionParams_t'?
   gap.setPreferredConnectionParams(&connectionParams);
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
       PreferredConnectionParams_t
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:111:37: error: 'connectionParams' was not declared in this scope
   gap.setPreferredConnectionParams(&connectionParams);
                                     ^~~~~~~~~~~~~~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:111:37: note: suggested alternative: 'connection_time'
   gap.setPreferredConnectionParams(&connectionParams);
                                     ^~~~~~~~~~~~~~~~
                                     connection_time
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:120:19: error: 'GapAdvertisingData' has not been declared
         .setFlags(GapAdvertisingData::BREDR_NOT_SUPPORTED
                   ^~~~~~~~~~~~~~~~~~
/Users/bsperryn/Documents/Arduino/libraries/Mbed_BLE_HID/src/Mbed_BLE_HID.cpp:121:19: error: 'GapAdvertisingData' has not been declared
                 | GapAdvertisingData::LE_GENERAL_DISCOVERABLE
                   ^~~~~~~~~~~~~~~~~~
exit status 1
Error compiling for board Arduino Nano 33 BLE.
```

--- a/README.md
+++ b/README.md
@@ -6,7 +6,7 @@ It was designed for the **Arduino nano 33 BLE** and tested with _GNU/Linux, Andr
 
 ## Environment
 
-On the Arduino IDE you will need the **Arduino nRF528x boards (Mbed OS)** with version **1.1.6** (In the menu bar click on "_Tools > Boards > Boards manager.._").
+On the Arduino IDE you will need the **Arduino mbed-enabled Boards** with version **1.3.1** or higher (In the menu bar click on "_Tools > Boards > Boards manager.._").
 
 Alternatively you can use [platformio](https://github.com/platformio) [Deviot](https://github.com/platformio/Deviot).
 

--- a/src/Mbed_BLE_HID.cpp
+++ b/src/Mbed_BLE_HID.cpp
@@ -102,14 +102,6 @@ void MbedBleHID::postInitialization(BLE &ble)
   Gap &gap = ble.gap();
   gap.setEventHandler(this);
 
-  const Gap::ConnectionParams_t connectionParams = {
-    7,          // min conn interval
-    15,         // max conn interval
-    0,          // slave latency
-    3200        // supervision timeout
-  };
-  gap.setPreferredConnectionParams(&connectionParams);
-
   // GAP Advertising parameters.
   {
     using namespace ble;
@@ -117,8 +109,8 @@ void MbedBleHID::postInitialization(BLE &ble)
     gap.setAdvertisingPayload(
       LEGACY_ADVERTISING_HANDLE,
       AdvertisingDataSimpleBuilder<LEGACY_ADVERTISING_MAX_SIZE>()
-        .setFlags(GapAdvertisingData::BREDR_NOT_SUPPORTED 
-                | GapAdvertisingData::LE_GENERAL_DISCOVERABLE
+        .setFlags(adv_data_flags_t::BREDR_NOT_SUPPORTED 
+                | adv_data_flags_t::LE_GENERAL_DISCOVERABLE
         )
         .setName(kDeviceName_.c_str(), true)
         .setAppearance(services_.hid->appearance())

